testing/deployer/sprawl/internal/build/docker.go

```diff
@@ -35,6 +35,64 @@ USER 100:0
 ENTRYPOINT []
 `
 
+const dockerfileDataplaneForTProxy = `
```

Comment:
This was a fun journey to concoct.

The gist is that the basic dataplane image is distroless, which makes a bunch of these modifications annoying so the very first step is to undo the last step of the dataplane image build (the distroless-ing part).

Then we ensure that `sudo`, `consul`, and `iptables` are present, and the uid `100` is named `consul` so that it can be added to the sudoers file.

Reply:
wow, this had to be fun!

nothing actionable / just to share context since we will eventually remove the consul container tests....

we currently use a dockerfile for this in consul-container tests. it was modified in https://github.com/hashicorp/consul/pull/19008/files#diff-b05b577992bd7304c4af2beaa76227ecd4f430bb06c299ea20e8ff5b20521439R14-R21 to allow tproxy tests to work, but that image has the drawbacks of using the ubi dataplane image and also the tests where tproxy is turned on don't run on Mac M1s.

I'm going to try the container tests with this definition locally. depending on when we think we should deprecate and remove consul-containers, we could also replace the above linked definition with what you have.  I'll reply with results, but nothing needed on your end.